### ISSUE TITLE
_physics_process delta doesn't change even under extreme lag.

### ISSUE THREAD

**Issue by Merlin1846** [godotengine/godot#93187]
### Tested versions

Reproducible in:
v3.5.1.stable.official [6fed1ffa3]
v4.2.2.stable.official [15073afe3]

### System information

Godot v4.2.2.stable - Ubuntu 23.10 23.10 - X11 - Vulkan (Forward+) - dedicated AMD Radeon RX 7600 (RADV GFX1102) () - AMD Ryzen 7 5800X 8-Core Processor (16 Threads)

### Issue description

`delta` is supposed to be the time between frames. When at 60 physics frames per second that's ~0.0166. And that is what it is when physics/common/physics_ticks_per_second is set to 60. However, if you say perform 1 million square roots per frame, this number should increase to match the fact that you are trying to set your computer on fire. However, it doesn't and instead stays at ~0.0166 regardless.

`_physics_process delta` should be constant, when the frame rate is constant. Otherwise, if the game slows down because someone's computer can't handle part of it, then the game will start to move in slow motion. And delta should increase, this will cause things like movement to take larger steps and as a result, someone running a game with physics/common/physics_ticks_per_second set to 60 when their computer can only handle 30 won't be stuck running at 50% the speed of time and instead will simply have less smooth motion visually (although it looks like smooth visuals is coming to 3d soon [https://github.com/godotengine/godot/pull/92391](https://github.com/godotengine/godot/pull/92391))

This could be fixed by lowering the physics framerate but now it needs to be increased/decreased based on how powerful the player's computer is, despite the fact that in most game engines it stays at 60 or even 30 regardless of what your computer can handle and keeping things running at the right speed is done by increasing delta when there is lag, however, this isn't being done for the physics but is for the graphics?

It appears that delta is just being set to `1 / Engine.physics_ticks_per_second` when it should change based on how long it has been since the last tick. With it being set to a fixed value then you can simply ignore delta and just keep the max physics frame rate set to 60 and no one will ever notice because even if their game lags it will behave **exactly** the same as if it was being multiplied by delta, that is to say, their game will run in slow motion.

### Steps to reproduce

Add this to something and you'll be able to see it in the console log and on screen. If you disable the for loops with a comment then the object moves much faster, but `_physics_process delta` doesn't change value to match, unlike `_process delta`.

```gdscript
func _process(delta: float) -> void:
	#position.x += 1000 * delta
	printt(delta, "A")
	#for i in 10000000:
		#1-1

func _physics_process(delta: float) -> void:
	position.x += 1000 * delta
	printt(delta, "b")
	
	for i in 10000000:
		1-1
```

### Minimal reproduction project (MRP)

[Archive.zip](https://github.com/user-attachments/files/15846036/Archive.zip)

**Comment by smix8** [comment]
> And delta should increase

No it should not. That is not how it works. Physics gets increasingly less accurate and more approximate as the time-step goes up or when steps are tiny small.

This is why it's integrated consistently at a small time-step in basically all common game engines. You would get an endless amount of physics "bugs" and unexpected results with such high or low deltas.

That is why the delta amount for physics steps is fixed and instead the physics step runs in a loop between 0 and up to 8 iterations in a frame depending on time passed. The 8 is an arbitrary cap that exists because inexperienced users would regularly attempt to fry their hardware when uncapped.

If a player runs only at 50% speed while the frame rate is low that is a scripting and usage issue on the user end, not a physics delta issue. The _physics_process() will be called 2+ times in that case so the only way to have such a slow movement is a scripting error that does not use a stepped movement.

**Comment by Merlin1846** [comment]
I did some more testing and it seems that based on what you said and my testing, that I misunderstood the way _physics_process was being called. I thought that it was being called once per physics frame and that the physics ticks which defaults to 0-8 was an internal thing that had to do with the way motion and collision was handled and not just it repeating the physics call.

I thought that it was functioning like a solver step where each tick was another iteration on the solver like with some simpler collision systems.

I hadn't found anyone saying that it was calling _physics_process() repeatedly so I thought was called once. Thank you, I adjusted(simplified) my code based on that and I'm getting consistent speeds.
